Chatto/sources/ChatController/ChatMessages/ChatMessagesViewController.swift

```diff
@@ -0,0 +1,427 @@
+
+        self.messagesAdapter.stopProcessingUpdates()
+    }
+
+    public override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+
+        if self.isFirstLayout {
+            self.messagesAdapter.startProcessingUpdates()
+            self.autoLoadMoreContentIfNeeded()
+        }
+        self.isFirstLayout = false
+    }
+
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        guard self.isViewLoaded else { return }
+        guard let collectionView = self.collectionView else { return }
+
+        let shouldScrollToBottom = self.collectionView.isScrolledAtBottom()
+        let referenceIndexPath = collectionView.indexPathsForVisibleItems.first
+        let oldRect = self.rect(at: referenceIndexPath)
+
+        coordinator.animate { [weak self] _ in
+            guard let self = self else { return }
+
+            if shouldScrollToBottom {
+                self.collectionView.scrollToBottom(
+                    animated: false,
+                    animationDuration: self.style.updatesAnimationDuration
+                )
+
+                return
+            }
+
+            let newRect = self.rect(at: referenceIndexPath)
+            self.collectionView.scrollToPreservePosition(oldRefRect: oldRect, newRefRect: newRect)
+        }
+    }
+
+    private func configureMessageAdapter() {
+        self.messagesAdapter.delegate = self
+        self.messagesAdapter.setup(in: self.collectionView)
+    }
+
+    private func configureStyle() {
+        self.collectionView.allowsSelection = false
+        self.collectionView.alwaysBounceVertical = self.style.alwaysBounceVertical
+        self.collectionView.backgroundColor = self.style.backgroundColor
+        self.collectionView.contentInset = self.style.contentInsets
+        self.collectionView.keyboardDismissMode = self.style.keyboardDismissMode
+        self.collectionView.showsHorizontalScrollIndicator = false
+        self.collectionView.showsVerticalScrollIndicator = self.style.shouldShowVerticalScrollView
+        self.collectionView.scrollIndicatorInsets = self.style.scrollIndicatorInsets
+    }
+
+    private func configureView() {
+        self.collectionView.chatto_setContentInsetAdjustment(enabled: false, in: self)
+        self.collectionView.chatto_setAutomaticallyAdjustsScrollIndicatorInsets(false)
+        self.collectionView.chatto_setIsPrefetchingEnabled(false)
+    }
+
+    private func configureCollectionView() {
+        self.collectionView.delegate = self
+    }
+
+    private func rect(at indexPath: IndexPath?) -> CGRect? {
+        guard let collectionView = self.collectionView else { return nil }
+        guard let indexPath = indexPath else { return nil }
+
+        return collectionView.collectionViewLayout.layoutAttributesForItem(at: indexPath)?.frame
+    }
+}
+
+// Proxy
```

Comment:
Should we have `autoLoadMoreContentIfNeeded` and `scroll(toItemId:postion:animated:spotlight)` in this extension as well?
They look like a usual implementation rather than proxy